lib/src/DynamicProto.ts

```diff
@@ -18,6 +18,36 @@ interface DynamicGlobalSettings {
 
 
 
+/**
+ * Helper to check if we're running in a restricted environment that doesn't support
+ * property redefinition, like Cloudflare Workers. This is primarily used to avoid
+ * operations that would cause issues in these environments.
+ * @ignore
+ */
+function _isRestrictedEnvironment(): boolean {
+    try {
+        // Test if we can perform property definition/redefinition
+        // This specifically targets restricted environments like Cloudflare Workers
+        // where property redefinition causes errors
+        let testObj = {};
+        let testProp = "testProperty";
+        Object.defineProperty(testObj, testProp, { 
```

Comment:
As this library doesn't use Object.defineProperty don't use this function, use an alternate method